### PR TITLE
Creategraph refactor

### PR DESCRIPTION
**Pull request by haholm**
**Changes:**
* Extract text_input and trace from create_graph to their own components.
* Move create_graph callbacks to controller.
* Move function for converting csv file to dataframe to utilities.
* Refactor and cleanup.

**I have:**

- [x] Set target branch to the correct branch, usually `dev`.
- [x] Rebased source branch from the target branch.
- [x] Ensured code is formatted and linted.
- [x] Cleaned up git history.
- [x] Ensured commit messages describe *what* was changed and *why*.
- [x] Ran tests locally after rebasing and checked output.

### CODE REVIEW

**Review by AxelMatstoms — COMMENTED** [review]
This should be a rebase.

File: src/dashboard/components/text_input.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+from typing import Any
+
+from dash import dcc
+
+
+def text_input(placeholder: str, **kwargs: Any) -> dcc.Input:
+    """Create a styled text input component.
+
+    Args:
+        placeholder: Placeholder text.
```
[the remark this answers is not in the document]
Comment (haholm) [reply src/dashboard/components/text_input.py]: Replaced text_input with existing one from dev.

File: src/dashboard/components/text_input.py
[the remark this answers is not in the document]
Comment (haholm) [reply src/dashboard/components/text_input.py:1]: Replaced text_input with existing one from dev.

File: src/dashboard/utilities.py [hunk 2]
```diff
@@ -117,3 +121,43 @@ def to_human_time_delta(duration: timedelta, abbreviated: bool = False) -> str:
...
+
+    Args:
+        contents: The string contents of the uploaded csv
+        files, each containing the file data type and data
+        seperated by a comma.
+
+    Returns: The polars dataframes, no entry is made if
+    contents is invalid.
+    """
+    loc_list = []
```
[the remark this answers is not in the document]
Comment (haholm) [reply src/dashboard/utilities.py]: Changed to `dataframes`.

File: src/dashboard/utilities.py [hunk 3]
```diff
@@ -117,3 +121,43 @@ def to_human_time_delta(duration: timedelta, abbreviated: bool = False) -> str:
...
+        by a comma.
+
+    Returns: The polars dataframe if the file contents is
+    valid, else None.
+    """
+    content_type, contents_data = contents.split(",")
+    try:
+        decoded = base64.b64decode(contents_data)
+    except binascii.Error:
+        return None
```
[the remark this answers is not in the document]
Comment (haholm) [reply src/dashboard/utilities.py]: Errors now slip through and are handled in `convert_to_dataframes`.

**Review by AxelMatstoms — APPROVED** [review]

**Pull request merged**